libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -36,6 +36,13 @@ export class BookSearchComponent implements OnInit {
     this.store.select(getAllBooks).subscribe((books) => {
       this.books = books;
     });
+
+    this.searchForm
+      .get('term')
+      .valueChanges.pipe(debounceTime(1000))
```

Comment:
This will again send API request if the value stays the same (rapidly hit a character and backspace). There is another RxJS operator which helps to avoid this. Please check